packages/@react-aria/table/src/useTableColumnResize.ts

```diff
@@ -28,37 +29,97 @@ export interface TableColumnResizeAria {
 }
 
 export interface AriaTableColumnResizeProps<T> {
+  /** An object representing the [column header](https://www.w3.org/TR/wai-aria-1.1/#columnheader). Contains all the relevant information that makes up the column header. */
   column: GridNode<T>,
+  /** Aria label for the hidden input. Gets read when resizing. */
   label: string,
-  triggerRef: RefObject<HTMLDivElement>,
+  /**
+   * Ref to the trigger if resizing was started from a column header menu. If it's provided,
+   * focus will be returned there when resizing is done.
+   * */
+  triggerRef?: RefObject<HTMLDivElement>,
+  /** If resizing is disabled. */
   isDisabled?: boolean,
-  onMove: (e: MoveMoveEvent) => void,
-  onMoveEnd: (e: MoveEndEvent) => void
+  /** If the resizer was moved. Different from onResize because it is always called. */
+  onMove?: (e: MoveMoveEvent) => void,
+  /**
+   * If the resizer was moved. Different from onResizeEnd because it is always called.
+   * It also carries the interaction details in the object.
+   * */
+  onMoveEnd?: (e: MoveEndEvent) => void,
+  /** Called when resizing starts. */
+  onResizeStart: (key: Key) => void,
+  /** Called for every resize event that results in new column sizes. */
+  onResize: (widths: Map<Key, number | string>) => void,
+  /** Called when resizing ends. */
+  onResizeEnd: (key: Key) => void
```

Comment:
Right, I was more wondering if `onMoveStart` (which doesn't exist here), `onMove`, and `onMoveEnd` could handle the logic from `onResizeStart`, `onResize`, and `onResizeEnd` as well. Now that I type this up, I now realize that the keyboard experience would always fire `onMove` -> `onMoveEnd` per keypress instead of how `onResizeEnd` only happens when finalizing a resize operation, so I think we are fine here

Reply:
I'd love to combine them, but I haven't figured out a good way to. Nor have I really figured out a better name for the difference.

Reply:
TODO: add to followup